stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelStateLogicImpl.kt

```diff
@@ -466,9 +466,8 @@ internal class ChannelStateLogicImpl(
 
         updateDataFromChannel(
             channel,
-            shouldRefreshMessages = request.isFilteringAroundIdMessages() ||
-                (!request.isFilteringMessages() && !mutableState.insideSearch.value),
-            scrollUpdate = request.isFilteringNewerMessages() || request.filteringOlderMessages()
+            shouldRefreshMessages = request.shouldRefresh,
```

Comment:
```suggestion
            shouldRefreshMessages = request.shouldRefresh || !request.isFilteringMessages(),
```

Reply:
Let's then move it to `watch` so we have a simple logic here.